pkg/controllerutil/resource_wrapper.go

```diff
@@ -109,6 +110,20 @@ func (r *ResourceFetcher[T]) ClusterVer() *T {
 func (r *ResourceFetcher[T]) ClusterComponent() *T {
 	return r.Wrap(func() (err error) {
 		r.ClusterComObj = r.ClusterObj.Spec.GetComponentByName(r.ComponentName)
+		if r.ClusterComObj != nil {
+			return
+		}
+		for _, shardingSpec := range r.ClusterObj.Spec.ShardingSpecs {
```

Comment:
it would be better to optimize and prevent the intrusion of sharding logic here

Reply:
> it would be better to optimize and prevent the intrusion of sharding logic here

refactor done